modules/creating-manifest-file-customized-br-ex-bridge.adoc

```diff
@@ -0,0 +1,127 @@
+
+Consider the following use cases for creating manifest object that includes a customized `br-ex` bridge:
+
+* You want to make postinstallation changes to the bridge, such as changing the Open vSwitch (OVS) or OVN-Kubernetes `br-ex` bridge network. The `configure-ovs.sh` shell script does not support making postinstallation changes to the bridge.
+* You want to deploy the bridge on a different interface than the interface available on a host or server IP address.
+* You want to make advanced configurations to the bridge that are not possible with the `configure-ovs.sh` shell script. Using the script for these configurations might result in the bridge failing to connect multiple network interfaces and facilitating data forwarding between the interfaces.
+
+ifndef::postinstall-bare-metal-ipi,postinstall-bare-metal-upi[]
+
+[NOTE]
+====
+If you require an environment with a single network interface controller (NIC) and default network settings, use the `configure-ovs.sh` shell script.
+====
+
+After you install {op-system-first} and the system reboots, the Kubernetes NMState Operator injects Ignition configuration files into each node in your cluster, so that each node received the `br-ex` bridge network configuration. The Operator prevents `br-ex` bridge configuration conflicts by sending a signal to the `configure-ovs.sh` shell script to not configure the `br-ex` bridge.
+endif::postinstall-bare-metal-ipi,postinstall-bare-metal-upi[]
+
+.Prerequisites
+
+
+.Procedure
+ifndef::postinstall-bare-metal-ipi,postinstall-bare-metal-upi[]
+. Create a `MachineConfig` manifest file and define a customized `br-ex` bridge network configuration similiar to the following example:
++
+[source,yaml]
+----
+apiVersion: machineconfiguration.openshift.io/v1
+kind: MachineConfig
+metadata:
+  labels:
+    machineconfiguration.openshift.io/role: worker <1>
+  name: 10-br-ex-worker
+spec:
+  config:
+    ignition:
+      version: 3.2.0
+    storage:
+      files:
+      - contents: <2>
+          source: <base64-encoded_access_key_id>
+          mode: 0644
+          overwrite: true
+          path: /etc/nmstate/openshift/<hostname>.yaml
+# ...
+----
+<1> Specify the `NetworkManager` connection name for the primary network interface.
+<2> For each note in your cluster, specify the hostname path to your node and the base-64-encoded Ignition configuration file data for the machine type.
+endif::postinstall-bare-metal-ipi,postinstall-bare-metal-upi[]
+ifdef::postinstall-bare-metal-ipi,postinstall-bare-metal-upi[]
+. Create a `NodeNetworkConfigurationPolicy (NNCP)` manifest file and define a customized `br-ex` bridge network configuration similiar to the following example:
++
+[source,yaml]
+----
+apiVersion: nmstate.io/v1
+kind: NodeNetworkConfigurationPolicy
+metadata:
+  name: br1-eth1-policy <1>
+spec:
+  desiredState:
+    interfaces:
+      - name: br1 <2>
+        description: Linux bridge with eth1 as a port <3>
+        type: linux-bridge <4>
```

Comment:
This is not the right example NMState config. We need an ovs-bridge for this.

There are some example files linked in https://issues.redhat.com/browse/OPNET-478 that provide the full needed NMState YAML.

Reply:
:heavy_check_mark: